### PR TITLE
Added rotary test, Disable and restore current profile while testing

### PR DESCRIPTION
**Pull request by marcos-diaz**

### CODE REVIEW

File: src/headers/profile.h [hunk 1]
```diff
@@ -74,5 +75,6 @@ Profile profile_init_console();
 Profile profile_init_console_legacy();
 Profile profile_init_desktop();
 
+extern uint8_t profile_active_index;
```
Comment (tomas-pecserke) [inline src/headers/profile.h]: This breaks encapsulation principle - it bad practice to expose implementation variable (and for write access no less).

There already is `void profile_set_active(uint8_t)` function, let's add a get function to go with it.

File: src/headers/rotary.h [hunk 2]
```diff
@@ -18,3 +18,5 @@ Rotary Rotary_ (
 );
 
 void rotary_init();
+
+extern int8_t rotary_increment;
```
Comment (tomas-pecserke) [inline src/headers/rotary.h]: Same here. Let's not expose implementation details like this if headers.

File: src/self_test.c [hunk 3]
```diff
@@ -10,98 +10,88 @@
...
-
     thumbstick->reset(thumbstick);
 }
 
-void self_test_rotary(Rotary* rotary) {
-    printf("Testing Rotary Wheel\n");
-
-    // TODO test rotary wheel
+void self_test_rotary_direction(const char *name, int8_t direction) {
+    rotary_increment = 0;
```
Comment (tomas-pecserke) [inline src/self_test.c]: this could use `rotaty.reset()` instead

**Review by tomas-pecserke — CHANGES_REQUESTED** [review]

File: src/headers/profile.h (hunk 1 above)
Comment (marcos-diaz) [reply src/headers/profile.h]: True I will make some set/get funcs

**Comment by tomas-pecserke** [comment]
A note on commit message format:

1) It's better to split functionality into separate commits for better readability of commit history.
1) It's preferable to use present tense in commit messages for better readability of commit history.

This way you read what each commit does, not what the developer did. Focus is on the commit itself.

Therefore

- `Disable profile before testing and restore afterwards`
- `Add rotary test`

reads much better than

- `Added rotary test, Disable and restore current profile while testing`

**Review by tomas-pecserke — APPROVED** [review]

**Pull request merged**